yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,334 @@
+"""
+Chimera data structures
+
+
+
+"""
+
+# -----------------------------------------------------------------------------
+# Copyright (c) 2013, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# -----------------------------------------------------------------------------
+
+import os
+
+import numpy as np
+
+from yt.geometry.geometry_handler import YTDataChunk
+from yt.geometry.unstructured_mesh_handler import UnstructuredIndex
+from yt.utilities.file_handler import HDF5FileHandler
+from yt.utilities.io_handler import io_registry
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+try:
+    from yt.data_objects.unstructured_mesh import SemiStructuredMesh
+except ModuleNotFoundError:
+    from yt.data_objects.index_subobjects.unstructured_mesh import SemiStructuredMesh
+
+
+from yt.data_objects.static_output import Dataset
+
+from .fields import ChimeraFieldInfo
+
+
+class ChimeraMesh(SemiStructuredMesh):
+
+    _index_offset = 0
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+def _find_files(
+    filename_c,
+):  # Returns a list of all files that share a frame number with the input
+    dirname, file = os.path.split(filename_c)
+    dir_contents = os.listdir(dirname)
+    frames = [f for f in dir_contents if file[:-7] in f]
+    directory = os.path.dirname(filename_c)
+    index_filenames = [directory + "/" + f for f in frames]
```

Comment:
windows compatibility !
```suggestion
    index_filenames = [directory + os.path.sep + f for f in frames]
```